### PR TITLE
[DRR] replace cc_test_old with cc_test

### PR DESCRIPTION
**Pull request by gongshaotian**
### PR types

Others
### PR changes

Others
### Description

DRR 测试：使用cc_test 替换cc_test_old
+ 修复编译优化导致的链接错误

Pcard-67164

### CODE REVIEW

File: test/cpp/pir/pattern_rewrite/CMakeLists.txt [hunk 1]
```diff
@@ -10,46 +10,29 @@ endif()
...
-  gtest
-  pd_op_dialect
-  pir)
+  SRCS drr_attention_fuse_test.cc
+  DEPS fusion_passes drr gtest pd_op_dialect pir)
 
 set_tests_properties(
   pattern_rewrite_test PROPERTIES ENVIRONMENT
                                   "FLAGS_enable_new_ir_in_executor=true")
+# set_source_files_properties(
```
Comment (yuanlehome) [inline test/cpp/pir/pattern_rewrite/CMakeLists.txt]: 注释删掉

Comment (gongshaotian) [reply test/cpp/pir/pattern_rewrite/CMakeLists.txt]: ok

**Review by yuanlehome — APPROVED** [review]
LGTM

**Review by risemeup1 — APPROVED** [review]
LGTM

**Pull request merged**